internal/pkg/aws/acm/acm.go

```diff
@@ -66,23 +70,33 @@ func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	}
 	for _, alias := range aliases {
 		if !validAliases[alias] {
-			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+			return &errInValidAliasAgainstCert{
+				certs:         certs,
+				alias:         alias,
+				domainsOfCert: domainsOfCert,
+			}
 		}
 	}
 	return nil
 }
 
-func (a *ACM) findValidAliasesAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, error) {
+func (a *ACM) validDomainsOfCert(ctx context.Context, cert string) ([]string, error) {
 	resp, err := a.client.DescribeCertificateWithContext(ctx, &acm.DescribeCertificateInput{
 		CertificateArn: aws.String(cert),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("describe certificate %s: %w", cert, err)
 	}
-	domainSet := make(map[string]bool)
-	domainSet[aws.StringValue(resp.Certificate.DomainName)] = true
-	for _, san := range resp.Certificate.SubjectAlternativeNames {
-		domainSet[aws.StringValue(san)] = true
+	var domainsOfCert []*string
+	domainsOfCert = append(domainsOfCert, resp.Certificate.DomainName)
```

Comment:
Do you think we need this line then? Based on the discussion here https://github.com/aws/copilot-cli/pull/4250#discussion_r1053572593

Reply:
removed that line as appending of canonical Domain name is not needed